hyperdome_server/web/share_mode.py

```diff
@@ -65,123 +66,114 @@ def __init__(self, common, web):
-        @self.web.app.route("/therapy_complete", methods=['POST'])
-        def therapy_complete():
-            self.connected_therapist.pop(
-                self.connected_guest[current_user.username])
-            self.connected_guest.pop(current_user.username)
-            self.therapists_available.append(current_user)
-
-        @self.web.app.route("/therapist_signout", methods=["POST"])
-        @login_required
-        def therapist_signout():
-            user = load_user(request.form['username'])
-            logout_user(user)
-            self.therapists_available.remove(user)
-
-        @self.web.app.route("/therapist_signup", methods=["POST"])
-        def therapist_signup():
-            if request.form.get('masterkey', "") == "megumin":
-                if load_user(request.form['username']):
-                    return "Username already exists"
-                user = self.user_class(username=request.form['username'],
-                                       password=request.form['password'])
-                self.db.session.add(user)
-                self.db.session.commit()
-                return "Success"
-            return abort(401)
-
-        @self.web.app.route("/therapist_signin", methods=["POST"])
-        def therapist_signin():
-            # TODO authenticate
-            user = load_user(request.form['username'])
-            self.therapists_available.append(user)
+            counselors = [
+                counselor for counselor in self.counselors_available if self.counselors_available[counselor] > 0]
+            if not counselors:
+                return ''
+            chosen_counselor = random.choice(counselors)
+            self.counselors_available[chosen_counselor] -= 1
+            self.active_chat_user_map[guest_id] = chosen_counselor
+            self.active_chat_user_map[chosen_counselor] = guest_id
+            return 'Success'
+
+        @self.web.app.route("/counseling_complete", methods=['POST'])
+        def counseling_complete():
+            sid = request.form['user_id']
+            if sid not in self.active_chat_user_map:
+                return 'no active chat', 404
+            other_user = self.active_chat_user_map[sid]
+            self.active_chat_user_map.pop(sid)
+            self.pending_messages.pop(sid, '')
+            self.active_chat_user_map.update({other_user: ''})
+            if sid in self.counselors_available:
+                self.counselors_available[sid] += 1
+            elif other_user in self.counselors_available:
+                self.counselors_available[other_user] += 1
+            return 'Chat Ended'
+
+        @self.web.app.route("/counselor_signout", methods=["POST"])
+        def counselor_signout():
+            sid = request.form['user_id']
+            self.counselors_available.pop(sid, '')
             return "Success"
 
+        @self.web.app.route("/counselor_signin")
+        def counselor_signin():
+            # TODO authenticate
```

Comment:
smh what are you gonna do if the hackers don't respect this demilitarized zone

Reply:
they can't do that, it's illegal.